includes/wikia/Wikia.php

```diff
@@ -1341,20 +1341,22 @@ static public function isUnsubscribed( $to, $body, $subject ) {
 		return true;
 	}
 
-	static public function allowNotifyOnPageChange ( /* User */ $editor, /* Title */ $title ) {
-		global $wgWikiaBotUsers;
-
-		$allow = true;
-		if ( !empty( $wgWikiaBotUsers ) ) {
-			foreach ( $wgWikiaBotUsers as $type => $user ) {
-				if ( $user["username"] == $editor->getName() ) {
-					$allow = false;
-					break;
-				}
-			}
-		}
+	/**
+	 * Do not send watchlist emails for edits made by Wikia bot accounts
+	 *
+	 * @param User $editor
+	 * @param Title $title
+	 * @return bool return false if you want to block an email
+	 */
+	static public function allowNotifyOnPageChange ( User $editor, /* Title */ $title ) {
+		global $wgWikiaBotLikeUsers;
 
-		return $allow;
+		if ( in_array( $editor->getName(), $wgWikiaBotLikeUsers)  ) {
```

Comment:
I'd keep the if / else approach as the "direct" `return` can be confusing - as the above proves ;)